src/librustc_yk_sections/emit_sir.rs

```diff
@@ -478,3 +479,28 @@ fn lower_def_id(tcx: TyCtxt<'_>, &def_id: &DefId) -> ykpack::DefId {
         def_idx: def_id.index.as_u32(),
     }
 }
+
+/// Was this MIR annotated with calls the the software tracer?
+/// We decide this by inspecting the terminator if the first block.
+fn was_annotated(tcx: TyCtxt<'_>, body: &Body<'_>) -> bool {
+    let first_block = match body.basic_blocks().iter().next() {
+        Some(b) => b,
+        None => return false, // No blocks. Couldn't have annotated this then.
+    };
+
+    if let TerminatorKind::Call{func: Operand::Constant(box Constant {
+        literal: Const {
+            ty: &TyS {
+                sty: ty::FnDef(ref def_id, _), ..
+            }, ..
+        }, ..
+    }, ..), ..} = first_block.terminator.as_ref().unwrap().kind {
+        // Block is terminated by a static call. So is it the trace recorder?
+        let rec_fn_defid = tcx.get_lang_items(LOCAL_CRATE).yk_swt_rec_loc()
+            .expect("couldn't find software trace recorder function");
+        return *def_id == rec_fn_defid;
+    }
+
+    // Block wasn't terminated by a static call.
+    return false;
```

Comment:
`return` isn't necessary here.

Reply:
true.

(well, no false, but true that I don't need the `return` 😆 )